setuptools/tests/test_bdist_wheel.py

```diff
@@ -0,0 +1,460 @@
+        assert set(wf.namelist()) == DEFAULT_FILES | license_files
+
+
+def test_licenses_deprecated(dummy_dist, monkeypatch, tmp_path):
+    dummy_dist.joinpath("setup.cfg").write_text(
+        "[metadata]\nlicense_file=licenses/DUMMYFILE", encoding="utf-8"
+    )
+    monkeypatch.chdir(dummy_dist)
+    subprocess.check_call([
+        sys.executable,
+        "setup.py",
+        "bdist_wheel",
+        "-b",
+        str(tmp_path),
+        "--universal",
+    ])
+    with WheelFile("dist/dummy_dist-1.0-py2.py3-none-any.whl") as wf:
+        license_files = {"dummy_dist-1.0.dist-info/DUMMYFILE"}
+        assert set(wf.namelist()) == DEFAULT_FILES | license_files
+
+
+@pytest.mark.parametrize(
+    "config_file, config",
+    [
+        ("setup.cfg", "[metadata]\nlicense_files=licenses/*\n  LICENSE"),
+        ("setup.cfg", "[metadata]\nlicense_files=licenses/*, LICENSE"),
+        (
+            "setup.py",
+            SETUPPY_EXAMPLE.replace(
+                ")", "  license_files=['licenses/DUMMYFILE', 'LICENSE'])"
+            ),
+        ),
+    ],
+)
+def test_licenses_override(dummy_dist, monkeypatch, tmp_path, config_file, config):
+    dummy_dist.joinpath(config_file).write_text(config, encoding="utf-8")
+    monkeypatch.chdir(dummy_dist)
+    subprocess.check_call([
+        sys.executable,
+        "setup.py",
+        "bdist_wheel",
+        "-b",
+        str(tmp_path),
+        "--universal",
+    ])
+    with WheelFile("dist/dummy_dist-1.0-py2.py3-none-any.whl") as wf:
+        license_files = {
+            "dummy_dist-1.0.dist-info/" + fname for fname in {"DUMMYFILE", "LICENSE"}
+        }
+        assert set(wf.namelist()) == DEFAULT_FILES | license_files
+
+
+def test_licenses_disabled(dummy_dist, monkeypatch, tmp_path):
+    dummy_dist.joinpath("setup.cfg").write_text(
+        "[metadata]\nlicense_files=\n", encoding="utf-8"
+    )
+    monkeypatch.chdir(dummy_dist)
+    subprocess.check_call([
+        sys.executable,
+        "setup.py",
+        "bdist_wheel",
+        "-b",
+        str(tmp_path),
+        "--universal",
+    ])
+    with WheelFile("dist/dummy_dist-1.0-py2.py3-none-any.whl") as wf:
+        assert set(wf.namelist()) == DEFAULT_FILES
+
+
+def test_build_number(dummy_dist, monkeypatch, tmp_path):
+    monkeypatch.chdir(dummy_dist)
+    subprocess.check_call([
+        sys.executable,
+        "setup.py",
+        "bdist_wheel",
+        "-b",
+        str(tmp_path),
+        "--universal",
+        "--build-number=2",
+    ])
+    with WheelFile("dist/dummy_dist-1.0-2-py2.py3-none-any.whl") as wf:
+        filenames = set(wf.namelist())
+        assert "dummy_dist-1.0.dist-info/RECORD" in filenames
+        assert "dummy_dist-1.0.dist-info/METADATA" in filenames
+
+
+def test_limited_abi(monkeypatch, tmp_path):
+    """Test that building a binary wheel with the limited ABI works."""
+    this_dir = os.path.dirname(__file__)
+    source_dir = os.path.join(this_dir, "testdata", "extension.dist")
```

Comment:
I don't think these files exist within the context of the PR.

Reply:
Yeah, I figured there might be something missing, but I was first distracted by failures unrelated to the wheel changes.

Reply:
The other errors are likely to be related to the accidental removal of `_vendor/backports/tarfile`

Reply:
It seems that the vendor command produced a different result on my machine than what was previously committed to the repo. I'll try to get rid of all the unrelated changes.

Reply:
Nice, now the CI failures are more easy to understand.

Reply:
BTW, I'm in room 315 now if you're at PyCon. Purple PyCon shirt, black Asus laptop.

Reply:
Sorry, I am not attending :(